join-domain/elx/sssd/files/join.sh

```diff
@@ -149,5 +150,55 @@ function JoinDomain {
   return ${RET_CODE}
 }
 
-IsDiscoverable
-JoinDomain
+# Try to leave and remove host from domain
+function LeaveDomain {
+  local    LEAVE_CRED
+  local -a REALM_OPTS
+
+  REALM_OPTS=(
+    -U "${JOIN_USER}"
+    --unattended
+    --remove
+  )
+
+  # Get credentials used for leave operation
+  LEAVE_CRED="$( PWdecrypt )"
+
+
+  printf "Removing %s from to %s" "$( hostname -s )" "${JOIN_DOMAIN}"
+
+  if [[ $(
+    echo "${LEAVE_CRED}" |
+    realm leave \
+      "${REALM_OPTS[@]}" \
+      "${JOIN_DOMAIN}" > /dev/null 2>&1
+  )$? -eq 0 ]]
+  then
+    RET_CODE=0
+
+    echo "Success"
+
+  else
+      echo "FAILED: Getting system logs"
+      printf "\n==============================\n"
+      journalctl -u realmd | \
+        grep "$( date '+%b %d %H:%M' )" | \
+        sed 's/^.*]: /: /'
+      printf "\n==============================\n"
+
+      RET_CODE=1
+  fi
+
+  exit "${RET_CODE}"
```

Comment:
This need to be a `return`?
```suggestion
  return "${RET_CODE}"
```

Reply:
Given where it is in the code, it's _technically_ incorrect but makes no _functional_ difference (script operates as expected when invoked either by hand or via Saltstac).